src/solvers/flattening/boolbv_quantifier.cpp

```diff
@@ -132,27 +128,25 @@ exprt get_quantifier_var_max(
   return res;
 }
 
-optionalt<exprt>
+static optionalt<exprt>
 instantiate_quantifier(const quantifier_exprt &expr, const namespacet &ns)
 {
-  PRECONDITION(expr.id() == ID_forall || expr.id() == ID_exists);
```

Comment:
This still sounds like a sensible precondition to me.

Reply:
It's already checked in `boolbvt::convert_quantifier` (which is the only caller of this `static` procedure), and will be checked again at the end of this procedure via an `UNREACHABLE` after handling `ID_forall` and `ID_exists`.

Reply:
OK.  If it were me I would leave it is as executable documentation but I agree that it is redundant so I think this is a matter of style and either is fine.